File: c#/#344_reverse_string.cs
using System;   //Length

public class Solution {
    public void ReverseString(char[] s) {
        int left = 0;
        int right = s.Length - 1;

        while (left < right){
            Swap(ref s[left], ref s[right]);
            ++left;
            --right;
        }
    }

    private void Swap(ref char a, ref char b) {
        char temp = a;
        a = b;
        b = temp;
    }
}

class Program {
    static void Main() {
        Solution sol = new Solution();

        char[] input = {'h', 'e', 'l', 'l', 'o'};
        Console.WriteLine("Original Array: " + new string(input));

        sol.ReverseString(input);
        Console.WriteLine("Reversed Array: " + new string(input));
    }
}

File: cpp/#344_reverse_string.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    void reverseString(vector<char>& s)
    {
        int left = 0;
        int right = s.size() - 1;

        while (left < right)
        {
            swap(s[left], s[right]);
            ++left;
            --right;
        }
    }
};

int main()
{
    Solution solution;
    vector<char> str = {'H', 'e', 'l', 'l', 'o', '!', '\0'};

    cout << "Original string: ";
    for (char c : str)
    {
        if (c != '\0')
        {
            cout << c;
        }
    }
    cout << endl;

    solution.reverseString(str);

    cout << "Reversed string: ";
    for (char c : str)
    {
        if (c != '\0')
        {
            cout << c;
        }
    }
    cout << endl;

    return 0;
}
